engine/types/type42/linking_key_test.go

```diff
@@ -0,0 +1,149 @@
+package type42
+
+import (
+	"encoding/hex"
+	"fmt"
+	"testing"
+
+	"github.com/libsv/go-bk/bec"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCalculateHMAC(t *testing.T) {
```

Comment:
The question is if we really want to test this function, as this is private function. And actually it is kinda internal helper for DeriveLinkedKey, I would rather suggest to delete this tests entirely and just focus on tests for DeriveLinkedKey.

Reply:
easier to remove it, so remove it .. as anyway latter in code we are testing DeriveLinkedKey and if this private method would fail this method will fail to `DeriveLinkedKey`